textile/chat-features.textile

```diff
@@ -408,35 +408,52 @@ Presence allows chat room users to indicate to others that they're online, as we
-** @(CHA-T2b)@ This specification point has been removed.
-*** @(CHA-T2b1)@ This specification point has been removed.
-*** @(CHA-T2b2)@ This specification point has been removed.
-** @(CHA-T2c)@ @[Testable]@ If the room status is @Attaching@, then the operation shall invoke a @CHA-RL9@ pending room status operation and wait for it to complete. If this operation fails, then the error from @CHA-RL9@ must be raised. If this succeeds (i.e the room becomes @ATTACHED@), the library must invoke the underlying @presence.get()@ call.
-** @(CHA-T2g)@ @[Testable]@ If the room status is anything other than @Attached@ or @Attaching@, an @ErrorInfo@ using the @RoomInInvalidState@ error code from the "chat-specific error codes":#error-codes with a status code of @400@ must be thrown. The message shall explain that attach must be called first.
-** @(CHA-T2d)@ @[Testable]@ If the room status is @Attached@, then the @get@ call shall invoke the underlying @presence.enter()@ call.
-** @(CHA-T2e)@ This specification point has been removed. It was superseded by @CHA-T2g@.
-** @(CHA-T2f)@ This specification point has been removed. It was superseded by @CHA-T2g@.
-* @(CHA-T3)@ @[Testable]@ Users may configure a timeout interval for when they are typing. This configuration is provided as part of the @RoomOptions@ @typing.timeoutMs@ property, or idiomatic equivalent. The default is 5000ms.
+For the purpose of this section, an @ephemeral message@ is understood to mean a regular message where @message.extras.ephemeral@ is set to @true@.
+
+* @(CHA-T1)@ This specification point has been removed. It was superseded by @CHA-T8@.
+* @(CHA-T2)@ This specification point has been removed. It was replaced by @CHA-T9@.
+* @(CHA-T8)@ Typing Indicators for a Room are exposed on the main room channel, in the format @<roomId>::$chat@. For example, if your room id is @my-room@ then the channel will be @my-room::$chat@.
+* @(CHA-T9)@ @[Testable]@ Users may retrieve a list of the currently typing client IDs.
+* @(CHA-T10)@ @[Testable]@ Users may configure a heartbeat interval (the period between typing heartbeats sent by the client). This configuration is provided at the @RoomOptions.typing.heartbeatThrottleMs@ property, or idiomatic equivalent. The default is 10000ms.
+** @(CHA-T10a)@ @[Testable]@ A grace period shall be set by the client (the grace period on the @CHA-T10@ heartbeat interval when receiving heartbeats). The value shall be set to 2000ms.
+*** @(CHA-T10a1)@ @[Testable]@ If a heartbeat is not received within this period, the client shall assume that the user has stopped typing. For example, if the client has not received a heartbeat within 12000ms of the last heartbeat (10s heartbeat interval plus 2s grace period), the client shall assume that the user has stopped typing.
 * @(CHA-T4)@ Users may indicate that they have started typing.
```

Comment:
Currently, we have 2 method for typing 
1. keystroke
2. stop
Both of those raise error if sending typing event fails. 
Ideally, typing event send is supposed to be an instantaneous process. 
So, if  `keystroke` and `stop` methods take more than 2 seconds to return a successful response, should we raise it as a timeout error? Ofc, we do check for channel attached state, so this case should be negligible, but just wanted to raise the point.

Reply:
```suggestion
* @(CHA-T4)@ Users may indicate that they have started typing using @keystroke@ method
```